src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/CheckPrintStatusTest.java

```diff
@@ -27,30 +44,129 @@
 @SpringBootTest
 public class CheckPrintStatusTest {
 
+    private static final String serviceName = "sendletterconsumer";
+
+    private static final LocalTime fivePm = LocalTime.of(17, 0, 0);
+
+    private static final LocalTime secondBeforeFivePm = LocalTime.of(16, 59, 59);
+
     @Autowired
     private MockMvc mvc;
 
+    @Autowired
+    private NamedParameterJdbcTemplate jdbcTemplate;
+
     @MockBean
     private AuthTokenValidator tokenValidator;
 
     @SpyBean
     private AppInsights insights; // NOPMD will be used with service implementation
 
     @SpyBean
-    private LetterRepository letterRepository; // NOPMD will be used with service implementation
+    private LetterRepository letterRepository;
+
+    @Before
+    public void setUp() {
+        given(tokenValidator.getServiceName("auth-header-value")).willReturn(serviceName);
+    }
+
+    @After
+    public void tearDown() {
+        jdbcTemplate.update("DELETE FROM letters", Collections.emptyMap());
+    }
+
+    @Test
+    public void should_return_204_when_there_are_no_unprinted_letters() throws Exception {
+        ArgumentCaptor<NotPrintedLetter> captor = ArgumentCaptor.forClass(NotPrintedLetter.class);
+
+        checkPrintStatus().andExpect(status().isNoContent());
+
+        verify(letterRepository).getStaleLetters();
+        // TODO
+        // verify(insights, never()).trackNotPrintedLetter(captor.capture());
+
+        assertThat(captor.getAllValues()).isEmpty();
+    }
+
+    @Test
+    public void should_return_204_when_there_are_two_not_printed_letters() throws Exception {
```

Comment:
I think we can split this test case into two `should_return_204_when_there_are_not_printed_letters_sent_before_five_pm` and `should_return_204_when_there_are_not_printed_letters_sent_at_five_pm`.Also I think we need to verify the arguments which is list of not printed letters in both these test cases.What do you think?

Reply:
was thinking of verifying the captured object, indeed. must've skipped this step alongside.

won't hurt to split the test too. this way have one "positive" test and the rest positives are representing failure in condition to be included in the report